nixos/modules/services/misc/osticket/default.nix

```diff
@@ -0,0 +1,282 @@
+{ config, lib, pkgs, ... }:
+
+let
+  version = "1.15.4";
+  sha256 = "1nic4sg7n1q3fr4sj2j7xgv3a5p1gq4xccmx5msci1wwx62mfjfw";
+  user = "osticket";
+  stateDir = "/var/lib/${user}";
+
+  inherit (lib) mkDefault mkEnableOption mkForce mkIf mkMerge mkOption;
+  inherit (lib)
+    concatStringsSep literalExpression mapAttrsToList optional optionals
+    optionalString types;
+  inherit (pkgs) fetchFromGitHub;
+
+  cfg = config.services.osticket;
+
+in
+{
+  # interface
+  options = {
+    services.osticket = {
+      enable = mkEnableOption "the osTicket issue tracker with mysql and phpfpm+nginx";
+
+      withSetup = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Include the setup wizard. Set this to false after setup.
+        '';
+      };
+
+      favicon = mkOption {
+        type = with types; nullOr path;
+        default = null;
+        description = ''
+          Path to custom favicon PNG image. This will be added to the page &lt;HEAD&gt; sections.
+        '';
+      };
+
+      # TODO make this nicer wrt getting the plugins, perhaps ship all plugins by default + allow extra
+      plugins = mkOption {
```

Comment:
It is possible this would be better as an argument to the derivation (when you move that to a separate file) than an option to the module... :thinking: Maybe `pkgs.osticket.withPackages (p: with p; [ slack ])`? :sweat_smile: